### DISCUSSION TITLE
VM encryption

### DISCUSSION THREAD

**Discussion by michabbs** [rfjakob/gocryptfs#853] [General]
I am going to use gocryptfs to encrypt qemu virtual machines. Is it bad idea?

Possible problem: the virtual disk is basically one very large file (5-200GB) accessed randomly. How gocryptfs works with random reads/writes in the middle of an encrypted file? (I hope it does not need to read the whole file, update content and then write everything back?)

**Comment by rfjakob** [comment]
Should work fine. Files are split into 4kiB chunks that can be read and written independently.
